### PR TITLE
*: refactor Executor.Next() to receive RecordBatch

### PR DESCRIPTION
**Pull request by lysu**
### What problem does this PR solve? 

Change `Executor#Next`'s method signature from

```
Next(ctx context.Context, chk *chunk.Chunk) error
```
to 

```
Next(ctx context.Context, req *chunk.RecordBatch) error
```

To make it easy to add other parameters to `Next`.

### What is changed and how it works?

only refactor, don't change any logic even though there are obvious optimize way exists, so it can make reviewer easier.

### Check List 

Tests 

 - Old test case

Code changes

 - Has exported function/method change

Side effects

 - N/A

Related changes

 - N/A

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/tidb/8994)

### CODE REVIEW

**Comment by lysu** [comment]
just refactor and easy to conflict with other PR, @zz-jason @XuHuaiyu PTAL thx

File: util/execution/req.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package execution
+
+import "github.com/pingcap/tidb/util/chunk"
+
+// ExecRequest is input parameter of Executor.Next` method.
+type ExecRequest struct {
```
Comment (winoros) [inline util/execution/req.go]: Should it put into `util` rather than `executor`?

Comment (lysu) [reply util/execution/req.go]: emm.. at first I had to try put it into executor, but I found this `req` not only used in executor but also some other package, so it doesn't belong to `executor` in strict, so at last I'm following `Chunk` does to put it here.

Comment (zz-jason) [reply util/execution/req.go]: can we put this struct into "util/execdetails"?

File: executor/ddl_test.go [hunk 2]
```diff
@@ -90,7 +91,7 @@ func (s *testSuite3) TestCreateTable(c *C) {
 	chk := rs.NewChunk()
 	it := chunk.NewIterator4Chunk(chk)
 	for {
-		err1 := rs.Next(ctx, chk)
+		err1 := rs.Next(ctx, &execution.ExecRequest{RetChunk: chk})
```
Comment (qw4990) [inline executor/ddl_test.go]: Suggest to use a method like `NewExecRequest(chk)` to create this struct.
If someone add a new field into `ExecRequest` with a specific default value, then he have to modify all places again.

File: util/execution/req.go (hunk 1 above)
Comment (zz-jason) [inline util/execution/req.go]: how about change the struct name to "RecordBatch" and make `Chunk` as it's embedded struct:
```go
type RecordBatch struct {
	*chunk.Chunk
}
```

In the outside, we can just call the function bound to `Chunk` directly though an object of `RecordBatch` like this:

```go
	var batch RecordBatch;
	batch.NumRows()
	batch.Reset()
	...
```

Comment (lysu) [reply util/execution/req.go]: emm, I have take care about this too, but lost info `Ret`Chunk but ok I can change .

Comment (lysu) [reply util/execution/req.go]: `execdetail` just like a collection info package...In my understand.... :rofl:

File: executor/ddl_test.go (hunk 2 above)
Comment (lysu) [reply executor/ddl_test.go]: I prefer 0 or false a default value, and this make more feeling like-named parameter, but add ctor is ok too... ok, let's use change

File: util/execution/req.go (hunk 1 above)
Comment (lysu) [reply util/execution/req.go]:
```
        imports github.com/pingcap/tidb/ddl
	imports github.com/pingcap/tidb/ddl/util
	imports github.com/pingcap/tidb/kv
	imports github.com/pingcap/tidb/util/execdetails
	imports github.com/pingcap/tidb/util/chunk
	imports github.com/pingcap/tidb/types
	imports github.com/pingcap/tidb/sessionctx/stmtctx
	imports github.com/pingcap/tidb/util/execdetails
```

Comment (lysu) [reply util/execution/req.go]: @zz-jason add to execdetails will got import cycle

Comment (lysu) [reply util/execution/req.go]: or let's put it into chunk package? @zz-jason  @winoros

Comment (winoros) [reply util/execution/req.go]: :joy: Oh `stmtctx` imports `execdeail`

Put it into `chunk` and name it `ChunkRequest`?

File: cmd/benchdb/main.go [hunk 3]
```diff
@@ -120,7 +121,7 @@ func (ut *benchDB) mustExec(sql string) {
 		rs := rss[0]
 		chk := rs.NewChunk()
 		for {
-			err := rs.Next(ctx, chk)
+			err := rs.Next(ctx, chunk.NewRecordBatch(chk))
```
[the remark this answers is not in the document]
Comment (lysu) [reply cmd/benchdb/main.go]: yes, just as PR description said " there are obvious optimize way exists" but this PR doesn't do that and just refactor to make review logic simple, maybe better do it in following per executor PR

[the remark this answers is not in the document]
Comment (zz-jason) [reply cmd/benchdb/main.go]: @lysu Could you make a list of refactor items you found in this PR? So that we can do it in the future PRs.

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**